### ISSUE TITLE
crash of tksidviewer.py on close

### ISSUE THREAD

**Issue by fenrog** [fenrog/supersid#25]
# Observation
```
(supersid_new) E:\work\Python-Samples\fenrog\supersid\supersid>python supersid.py
Config file '..\Config\supersid.cfg' read successfully
sounddevice device 'MME: Microsoft Sound Mapper - Input', sampling rate 192000, format S32_LE
- sounddevice 'MME: Microsoft Sound Mapper - Input'
self.capture_device.close()
Exception in thread Thread-1:
Traceback (most recent call last):
  File "C:\Users\Admin\.conda\envs\supersid_new\lib\threading.py", line 932, in _bootstrap_inner
    self.run()
  File "C:\Users\Admin\.conda\envs\supersid_new\lib\threading.py", line 1254, in run
    self.function(*self.args, **self.kwargs)
  File "E:\work\Python-Samples\fenrog\supersid\supersid\sidtimer.py", line 69, in _ontimer
    self.callback()
  File "supersid.py", line 173, in on_timer
    self.viewer.status_display(message, level=2)
  File "E:\work\Python-Samples\fenrog\supersid\supersid\tksidviewer.py", line 111, in status_display
    self.statusbar_txt.set(message)
  File "C:\Users\Admin\.conda\envs\supersid_new\lib\tkinter\__init__.py", line 377, in set
    return self._tk.globalsetvar(self._name, value)
RuntimeError: main thread is not in main loop
```

# How to reproduce
```
[PARAMETERS]
viewer = tk
```
Happens sometimes when closing the tk window.

**Comment by fenrog** [comment]
SidTimer._ontimer() is triggered by an expiring timer.
SidTimer._ontimer() calls self.callback().
self.callback() is effectively SuperSID.on_timer(), as this function has been registered as callback.
SuperSID.on_timer() calls self.viewer.status_display(), in this case tkSidViewer.status_display()
Due to a race condition a moment before tkSidViewer.close() has been called, destroying self.tk_root.
Now tkSidViewer.status_display() tries to display the statusbar of a now longer existing window.
The above Exception appears.

Solution:
Maintain a runing flag and set the status display only if the mianloop is running.
The same for wxSidViewer.

**Comment by fenrog** [comment]
fixed with pull request #26
